test/request-data/user-preferences/preferences.spec.js

```diff
@@ -0,0 +1,209 @@
+import sinon from 'sinon';
+import { watch } from 'vue';
+
+import UserPreferences from '../../../src/request-data/user-preferences/preferences';
+
+import createTestContainer from '../../util/container';
+import testData from '../../data';
+import { mockHttp } from '../../util/http';
+
+const setupPreferences = (data = {}) => {
+  const user = testData.extendedUsers.createNew({ preferences: data });
+  const container = createTestContainer();
+  const preferences = new UserPreferences(user.preferences, container);
+  return { container, preferences };
+};
+
+describe('UserPreferences', () => {
+  it('returns preferences that have been set', () => {
+    const { preferences } = setupPreferences({
+      site: { projectSortMode: 'alphabetical' },
+      projects: {
+        123: { formTrashCollapsed: true }
+      }
+    });
+    preferences.site.projectSortMode.should.equal('alphabetical');
+    preferences.projects[123].formTrashCollapsed.should.be.true;
+  });
+
+  it('returns an object for a project without preferences', () => {
+    const { preferences } = setupPreferences();
+    expect(preferences.projects[123]).to.be.an('object');
+  });
+
+  it('returns the same object each time for a project without preferences', () => {
+    const { preferences } = setupPreferences();
+    expect(preferences.projects[123]).to.equal(preferences.projects[123]);
```

Comment:
Using `expect()` because `preferences.projects[123]` is a proxy that prevents access to its `.should` property.